### PR TITLE
ENH: BTI tweaks and source space morphing work

### PR DESCRIPTION
**Pull request by Eric89GXL**
@agramfort can you test the example?

### CODE REVIEW

File: examples/forward/plot_source_space_morphing.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Authors: Denis A. Engemann <denis.engemann@gmail.com>
+#          Eric larson <larson.eric.d@gmail.com>
+#
+# License: BSD (3-clause)
```
Comment (agramfort) [inline examples/forward/plot_source_space_morphing.py]: put authors after doc header

File: mne/source_space.py [hunk 2]
```diff
@@ -2372,30 +2405,86 @@ def morph_source_spaces(src_from, subject_to, surf='white', subjects_dir=None,
...
+                                  hemi, subjects_dir)
+        full_mapping = best[src_from[ii]['vertno']]
+        # Tragically, we might not have all of our vertno left (e.g. because
+        # some are omitted during fwd calc), so we must do some indexing magic:
+
+        # From all vertices, a subset could be chosen by fwd calc:
+        used_vertices = np.in1d(full_mapping, vertices[ii])
+        from_vertices.append(src_from[ii]['vertno'][used_vertices])
+        remaining_mapping = full_mapping[used_vertices]
+        assert np.array_equal(np.sort(remaining_mapping), vertices[ii])
```
Comment (agramfort) [inline mne/source_space.py]: turns this into a test

File: mne/source_space.py [hunk 3]
```diff
@@ -2372,30 +2405,86 @@ def morph_source_spaces(src_from, subject_to, surf='white', subjects_dir=None,
...
+        # And our data have been implicitly remapped by the forced ascending
+        # vertno order in source spaces
+        implicit_mapping = np.argsort(remaining_mapping)  # happens to data
+        data_idx = np.argsort(implicit_mapping)  # to reverse the mapping
+        data_idx += offset  # hemisphere offset
+        data_idxs.append(data_idx)
+        offset += len(implicit_mapping)
+    data_idx = np.concatenate(data_idxs)
+    assert np.array_equal(np.sort(data_idx),
+                          np.arange(sum(len(v) for v in vertices)))
```
Comment (agramfort) [inline mne/source_space.py]: turns this into a test

File: mne/source_estimate.py [hunk 4]
```diff
@@ -1366,6 +1367,41 @@ def plot(self, subject=None, surface='inflated', hemi='lh',
         return brain
 
     @verbose
+    def return_to_original_src(self, src_orig, subject_orig=None,
```
Comment (agramfort) [inline mne/source_estimate.py]: I don't like this name. Maybe

remap_to_src ?
map_to_src ?
map_to ?

cc @dengemann @teonlamont anyone ?

Comment (Eric89GXL) [inline mne/source_estimate.py]: `map_back_to_src` is more explicit because it makes it clear it must have been mapped forward in the first place

Comment (dengemann) [inline mne/source_estimate.py]: @Eric89GXL @agramfort to me this is clearly remapping. We should also make sure that it's not confused with morphing as remapping just changes the vertnos while keeping the data + shape. We should call it `remap_vertno` or `remap_vertices`. I don't like `_to_src` as it implies that there are multiple choices, maybe `remap_original_vertices`.

Comment (dengemann) [inline mne/source_estimate.py]: More options:
- `to_original_src`
- `as_original_src`

File: mne/tests/test_source_space.py [hunk 5]
```diff
@@ -573,6 +573,69 @@ def test_morph_source_spaces():
...
+    assert_equal(stc_morph_return.subject, stc_morph_morph.subject)
+    for ii in range(2):
+        assert_array_equal(stc_morph_return.vertices[ii],
+                           stc_morph_morph.vertices[ii])
+    # These will not match perfectly because morphing pushes data around
+    corr = np.corrcoef(stc_morph_return.data[:, 0],
+                       stc_morph_morph.data[:, 0])[0, 1]
+    assert_true(corr > 0.99, corr)
+
+    # Degenerate cases
```
Comment (dengemann) [inline mne/tests/test_source_space.py]: nice!

File: mne/source_estimate.py (hunk 4 above)
Comment (dengemann) [inline mne/source_estimate.py]: ping @Eric89GXL @agramfort

File: mne/io/bti/bti.py [hunk 6]
```diff
@@ -1053,13 +1058,16 @@ class RawBTi(_BaseRaw):
     def __init__(self, pdf_fname, config_fname='config',
                  head_shape_fname='hs_file', rotation_x=0.,
                  translation=(0.0, 0.02, 0.11), convert=True,
-                 ecg_ch='E31', eog_ch=('E63', 'E64'), verbose=None):
+                 rename_channels=True, sort_by_ch_name=True,
+                 ecg_ch='E31', eog_ch=('E63', 'E64'),
```
Comment (agramfort) [inline mne/io/bti/bti.py]: you really want to hard code these values? Are they the HCP values?

Comment (agramfort) [inline mne/io/bti/bti.py]: forget it...

File: mne/source_space.py [hunk 7]
```diff
@@ -2372,30 +2405,91 @@ def morph_source_spaces(src_from, subject_to, surf='white', subjects_dir=None,
...
+        # Get the mapping from the original source space to the destination
+        # subject's surface vertex numbers
+        best = _get_vertex_map_nn(src_from[ii], subject_from, subject_to,
+                                  hemi, subjects_dir)
+        full_mapping = best[src_from[ii]['vertno']]
+        # Tragically, we might not have all of our vertno left (e.g. because
+        # some are omitted during fwd calc), so we must do some indexing magic:
+
+        # From all vertices, a subset could be chosen by fwd calc:
+        used_vertices = np.in1d(full_mapping, vertices[ii])
```
Comment (agramfort) [inline mne/source_space.py]: use in1d from fixes

File: mne/source_space.py [hunk 8]
```diff
@@ -2372,30 +2405,91 @@ def morph_source_spaces(src_from, subject_to, surf='white', subjects_dir=None,
...
+        from_vertices.append(src_from[ii]['vertno'][used_vertices])
+        remaining_mapping = full_mapping[used_vertices]
+        if not np.array_equal(np.sort(remaining_mapping), vertices[ii]) or \
+                not np.in1d(vertices[ii], full_mapping).all():
```
Comment (agramfort) [inline mne/source_space.py]: here too

File: mne/io/bti/bti.py (hunk 6 above)
Comment (agramfort) [inline mne/io/bti/bti.py]: maybe just a nitpick but I would have added the new params at the end to avoid breaking API.

**Comment by agramfort** [comment]
that's it for me

File: mne/io/bti/bti.py (hunk 6 above)
Comment (dengemann) [inline mne/io/bti/bti.py]: @agramfort these params were actually never passed to the consrtuctor.

Comment (dengemann) [inline mne/io/bti/bti.py]: And we should  not support orderd arguments for keyword arguments it is bad practice anyway.  Here it's more redable to have all conversion options in one block.

**Comment by agramfort** [comment]
+1 for merge when travis is happy.

update the what's new in master directly.

**Comment by agramfort** [comment]
travis is happy.

please merge and update what's new

**Comment by Eric89GXL** [comment]
Thanks @agramfort

**Pull request merged**